docs/markdown/FAQ.md

```diff
@@ -551,4 +551,81 @@ libA = library('libA', 'fileA.cpp', link_with : [])
 This means that you HAVE to write your `library(...)` calls in the order that the
 dependencies flow. While ideas to make arbitrary orders possible exist, they were
 rejected because reordering the `library(...)` calls was considered the "proper"
-way. See [here](https://github.com/mesonbuild/meson/issues/8178) for the discussion.
\ No newline at end of file
+way. See [here](https://github.com/mesonbuild/meson/issues/8178) for the discussion.
+
+## Why doesn't meson have user defined functions/macros?
+
+The tl;dr answer to this is that meson's design is focused on solving specific
+problems rather than providing a general purpose language to write complex
+code solutions in build files. Build systems should be quick to write and
+quick to understand, functions muddle this simplicity.
+
+The long answer is twofold:
+
+First, Meson aims to provide a rich set of tools that solve specific problems
+simply out of the box. This is similar to the "batteries included" mentality of
+Python. By providing tools that solve common problems in the simplest way
+possible *in* Meson we are solving that problem for everyone instead of forcing
+everyone to solve that problem for themselves over and over again, often
+badly. One example of this are Meson's dependency wrappers around various
+config-tool executables (sdl-config, llvm-config, etc). In other build
+systems each user of that dependency writes a wrapper and deals with the
+corner cases (or doesn't, as is often the case), in Meson we handle them
+internally, everyone gets fixes and the corner cases are ironed out for
+*everyone*. Providing user defined functions or macros goes directly against
+this design goal.
```

Comment:
This is a fine reason, but I would have added here that Meson has modules to wrap tools from popular projects like GNOME/QT5. Having those modules built-in Meson greatly reduce the need of doing custom functions too.